src/server/replica.cc

```diff
@@ -528,16 +590,23 @@ error_code Replica::InitiateDflySync() {
     return make_error_code(errc::bad_message);
   }
 
-  for (unsigned i = 0; i < num_df_flows_; ++i) {
-    shard_flows_[i]->sync_fb_.join();
+  VLOG(0) << "DflySync before block";
+  // wait for all fibers to arrive on fs cut
+  {
+    std::unique_lock lk(block.mu_);
+    block.cv_.wait(lk, [&](){
+      return block.events_left == 0;
+    });
   }
+  VLOG(0) << "DflySync after block";
 
   absl::Duration dur = absl::Now() - start;
   uint64_t ms = absl::ToInt64Milliseconds(dur);
   LOG(INFO) << "Full sync finished in " << ms << "ms";
 
   state_mask_ |= R_SYNC_OK;
 
+  // TODO: block ptr is invalid now. Not in use, but still around.
   return error_code{};
```

Comment:
Nothing wrong with it. Its just that if someone decides to access block one more time he'll segfault. Maybe use shared ptr?